src/duckstation-qt/main.cpp

```diff
@@ -103,6 +104,13 @@ int main(int argc, char* argv[])
 
   if (boot_params)
   {
+    // In Challenge mode, do not allow loading a save state under no circumstances
+    // If it's present, drop it
+    if (host_interface->IsCheevosChallengeModeActive())
+    {
+      boot_params->state_stream.reset();
+    }
```

Comment:
Perhaps this should be done in `bootSystem` to make it absolutely bulletproof? Other code paths shouldn't be passing the state stream to `bootSystem` but it could act as an additional safeguard.